### PR TITLE
snapstate: relax gadget constraints in ConfigDefaults Et al.

### PR DESCRIPTION
**Pull request by mvo5**
The ConfigDefaults and GadgetConnections current read the
gadget data and sets constraints there. There is no need
at this point to check the constraints. This should have
happend earlier already and just complicated this particular
code.

This will unblock seeding in "install" mode for UC20.

There is more todo - i.e. we need to ensure we check gadget
consistency, see https://github.com/snapcore/snapd/pull/7880#issuecomment-564062194

But having this should unblock UC20.

### CODE REVIEW

File: overlord/snapstate/snapstate_test.go [hunk 1]
```diff
@@ -12387,6 +12387,55 @@ func (s *snapmgrTestSuite) TestConfigDefaults(c *C) {
...
+          type: 83,0FC63DAF-8483-4772-8E79-3D69D8477DE4
+          # whats the appropriate size?
+          size: 750M
+        - name: ubuntu-data
+          role: system-data
+          filesystem: ext4
+          type: 83,0FC63DAF-8483-4772-8E79-3D69D8477DE4
+          size: 1G
+`)
+	// XXX: in reality this will be a UC20 model context
```
Comment (pedronis) [inline overlord/snapstate/snapstate_test.go]: can we avoid the XXX ?

File: overlord/snapstate/snapstate.go [hunk 2]
```diff
@@ -2374,11 +2374,8 @@ func ConfigDefaults(st *state.State, deviceCtx DeviceContext, snapName string) (
 		return nil, state.ErrNoState
 	}
 
-	constraints := &gadget.ModelConstraints{
-		Classic: release.OnClassic,
-	}
-
-	gadgetInfo, err := gadget.ReadInfo(info.MountDir(), constraints)
+	// no constraints set: those should have been checked before already
```
Comment (pedronis) [inline overlord/snapstate/snapstate.go]: s/set/enforced/

File: overlord/snapstate/snapstate.go [hunk 3]
```diff
@@ -2411,11 +2408,8 @@ func GadgetConnections(st *state.State, deviceCtx DeviceContext) ([]gadget.Conne
 		return nil, err
 	}
 
-	constraints := &gadget.ModelConstraints{
-		Classic: release.OnClassic,
-	}
-
-	gadgetInfo, err := gadget.ReadInfo(info.MountDir(), constraints)
+	// no constraints set: those should have been checked before already
```
Comment (pedronis) [inline overlord/snapstate/snapstate.go]: same

File: overlord/snapstate/snapstate_test.go (hunk 1 above)
Comment (mvo5) [reply overlord/snapstate/snapstate_test.go]: Sure, added this. It's a bit more work (make the PR longer) but it's the right thing.

**Pull request merged**